cpp/src_prims/selection/knn.h

```diff
@@ -104,93 +170,451 @@ void merge_tables(long n, long k, long nshard, float *distances, long *labels,
    * @param res_I      pointer to device memory for returning k nearest indices
    * @param res_D      pointer to device memory for returning k nearest distances
    * @param k        number of neighbors to query
+   * @param allocator the device memory allocator to use for temporary scratch memory
    * @param s the cuda stream to use
+   * @param internalStreams optional when n_params > 0, the index partitions can be
+   *        queried in parallel using these streams. Note that n_int_streams also
+   *        has to be > 0 for these to be used and their cardinality does not need
+   *        to correspond to n_parts.
+   * @param n_int_streams size of internalStreams. When this is <= 0, only the
+   *        user stream will be used.
+   * @param translations translation ids for indices when index rows represent
+   *        non-contiguous partitions
```

Comment:
knitpick: doxygen documentation of params is incomplete, contains obsolete params and also not in order.